### PR TITLE
Avoid unsoundness in the use of the Pin API

### PR DESCRIPTION
**Pull request by mystor**

### CODE REVIEW

**Review by yaahc — APPROVED** [review]
looks good to me, I wanna get an approval from kenny too before I merge

cc @kennytm

File: fakecore/src/any.rs
Comment (yaahc) [inline fakecore/src/any.rs:54]: I kinda wanna rename `is` to `is_requesting`

File: fakecore/src/any.rs
Comment (kennytm) [inline fakecore/src/any.rs]: i think we could keep using `Self` here 🙂

**Review by kennytm — APPROVED** [review]
LGTM

File: fakecore/src/any.rs
Comment (mystor) [inline fakecore/src/any.rs]:
```suggestion
        F: FnOnce(Pin<&mut Self>),
```

**Pull request merged**
